### DISCUSSION TITLE
[CLOSED] Switching between two folders of shortcut keys

### DISCUSSION THREAD

**Discussion by ineuw** [autokey/autokey#754] [Q&A]
What is the best solution for switching between two definition folders, because the keys must be the same, but the output is in a different language? By selecting individual folders from ~/.config/autokey/data within Autokey, and then restarting Autokey, would this affect the desired change?

**Comment by josephj11** [comment]
Nobody I know of does that often, so I don't know if there's a "best" way.

What I did (for switching between beta and production) was to make the desired folder a symlink (in my case, the whole $HOME/.config/autokey) and then I wrote a bash script to change where that symlink points.

You could use an AutoKey script to run such a script, but I would experiment with that cautiously because having it happen dynamically might confuse AutoKey. Theoretically, AutoKey watches its directories and updates when it notices external changes, but I don't think anyone has stress tested it to that extent.

So, terminating AutoKey, changing the config files (or symlinks) and then restarting AutoKey will definitely work. Doing it on the fly, as above, might work too.

You could also define an AutoKey global stored variable that tells you which language is active and have all your scripts use that to determine what they should do, but that won't affect phrases unless they all use script macros which is a bit extreme.

Note that swapping folders/symlinks can be done at the top level like I did or just for particular folders or subtrees.

All my script and phrase folders are in a subtree that is outside of the AutoKey config tree. This makes backing them up
 a bit easier and they are unaffected  if I remove or even purge the AutoKey package. I had to symlink and swap that too.

There may be another approach to this. @BlueDrink9 built this whole capability that I don't really understand. It loads a whole set of scripts, etc. temporarily making them active, but they don't persist between AutoKey executions. IDK if or how that might apply here.

**Reply by BlueDrink9** [reply]
Yeah, temporary hotkeys were very much designed for this modal kind of setup, but I haven't really tested or documented them well. The idea is that rather than defining hotkeys using the gui, you put them all in a python script defined by the engine api. You can create a hotkey that clears all temporary hotkeys and loads another script that creates different ones

**Comment by ineuw** [comment]
Symlink is a good idea. Thanks.

**Reply by ineuw** [reply]
For this, and various other language related issues, I installed a second LMC 211 in French. This was a no-brainer in a bilingual world. Each installation is on its own 128Gb SSD drive and they share the data drives and separate Autokey installations.

**Reply by josephj11** [reply]
That works, but makes you update everything twice - once on each install. It really shouldn't be necessary to do such things. You should be able to change your locale settings and have everything just work. Maybe the above symlink script could read the current locale value and use that to choose the correct symlink.

The Linux `locale` command returns info about the current locale and `localectl` lets you set it.

**Comment by ineuw** [comment]
Sorry, I posted the reply in the wrong thread. I will move it here.

I considered your concerns about managing duplicates, but they don't apply in this case. Working in a single language environment is a gratifying experience, depending on what one wants out of it.

I tried turning my desktop into a bilingual workstation, but after two years of experimentation and testing I gave up. The process started two years ago with Windows 10, but it was a mess. Linux Mint is less of a mess, but both are struggling with managing keyboard definitions.

I spent months of planning, starting with the Autokey hotkeys to eliminate dual function keys so that I can use the same hotkey in either language. That is my second nature.

My AK hotkeys don't use the "ctrl" keys and the Linux Mint hotkeys that include the  "super" (win) key were disabled. Alt, Shift and Super has sufficient combinations for my 72 scripts. 

The web sites of Wikimedia are founded on the same basic software but there are profound differences in looks, approach, and implementation. Being a very elegant language, the French Wikipedia and Wikisource reflects this in both code and presentation. While English Wikipedia is universal, English Wikisource reflects a heavy Australian "British empire" influence, because most active contributors are from Australia. 

For me, it's also how the two languages relate and diverge. A topic for another forum.

**Reply by ineuw** [reply]
This is no longer an issue. Can you please close it?

**Comment by josephj11** [comment]
Since this is a discussion and not an issue, I don't think there's a way to close it. I might be able to lock it so no further comments are possible or delete it entirely, but neither of those are good options.

I could possibly (create and) add a new label to it such as inactive or completed, but there is no AutoKey policy (not even an informal one) about labeling discussions, so I usually don't label them. Any thoughts?

This is one of the reasons I don't like GitHub discussions - other ones being that they share the same numbering sequence with issues and that most of our active users don't ever see them or participate because they don't look to GitHub for support.

**Comment by Elliria** [comment]
It looks like GitHub is aware of that and is working on it, but you still might want to jump in on [their #3097 discussion about it](https://github.com/orgs/community/discussions/3097).

**Reply by josephj11** [reply]
Thanks for making us aware of this! I just commented there and upvoted a bunch of posts.

There are not a lot of upvotes over there, so anyone interested in this should consider upvoting and posting there.

**Reply by Elliria** [reply]
I added my up-vote, too. I wasn't sure how to do it, so I hit the up arrow and the smiley face on the message from the developer.

**Comment by ineuw** [comment]
I added the word [CLOSED] to the title. That's the best I could do in addition to upvoting the topic.

Relating to the original topic of two folders for two languages  ... The topic was unnecessary. But I appreciate your pointing out the editing of two separate setups.

**Reply by josephj11** [reply]
I have no idea what this is about (other than the first sentence). But, since this discussion is done, it probably really doesn't matter.

**Reply by ineuw** [reply]
Corrected it. My bad, my pen forgot to catch up to my thoughts. Mea Culpa.
